js-src/src/app/services/game.service.js

```diff
@@ -13,8 +13,8 @@ var GameService = Class({
             }
         )
     },
-    getWaitingGames: function () {
-        return this.http.get(this.baseUrl + '?state=waitingForPlayers').map(
+    getWaitingGames: function (diff) {
```

Comment:
I think for get games and get all games, the result should be the specific one or all. When players choose a difficulty, they should get the same difficulty game as they chose.